driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -502,33 +526,45 @@ public void onSet(Connection connection, Message.Response response, long latency
                                 connection.release();
                                 // Try another node
                                 logger.warn("Host {} is overloaded, trying next host.", connection.address);
-                                DriverException overloaded = new DriverException("Host overloaded");
-                                logError(connection.address, overloaded);
                                 if (metricsEnabled())
                                     metrics().getErrorMetrics().getOthers().inc();
-                                retry(false, null);
-                                return;
+                                if (retryPolicy instanceof ClientFailureAwareRetryPolicy)
```

Comment:
The code 
```java
connection.release();
if (retryPolicy instanceof ExtendedRetryPolicy)
    retry = ((ExtendedRetryPolicy)retryPolicy).onUnexpectedError(statement,
        request().consistency(),
        (DriverException)exceptionToReport,
        retriesByPolicy);
else
    retry = RetryPolicy.RetryDecision.tryNextHost(null);
if (metricsEnabled()) {
    metrics().getErrorMetrics().getOthers().inc();
    if (retry.getType() == Type.RETRY)
        metrics().getErrorMetrics().getRetriesOnUnexpectedError().inc();
    if (retry.getType() == Type.IGNORE)
        metrics().getErrorMetrics().getIgnoresOnUnexpectedError().inc();
}
```
Is the same in each one of the new Exceptions exposed in the retry policy (overloaded/bootstrapped/server_error). I though it would be clearer to group this in a function because now the behaviour is the same everywhere, except for `SERVER_ERROR` where we also defunct the connection.